packages/core/contracts/common/implementation/AncillaryData.sol

```diff
@@ -33,4 +33,28 @@ library AncillaryData {
         if (uint8(b) < 10) return bytes1(uint8(b) + 0x30);
         else return bytes1(uint8(b) + 0x57);
     }
+
+    /**
+     * @notice Adds "key:value" to `currentAncillaryData` where `value` is an address that first needs to be converted
+     * to utf8 bytes. For example, if `utf8(currentAncillaryData)="k1:v1"`, then this function will return 
+     * `utf8(k1:v1,key:value)`, and if `currentAncillaryData` is blank, then this will return `utf8(key:value)`.
+     * @param currentAncillaryData This bytes data should ideally be able to be utf8-decoded, but its OK if not. 
+     * @param key Again, this bytes data should ideally be able to be utf8-decoded, but its OK if not. 
+     * @param value An address to set as the value in the key:value pair to append to `currentAncillaryData`.
+     * @return Newly appended ancillary data.
+     */
+    function appendAddressKey(bytes memory currentAncillaryData, bytes memory key, address value) internal pure returns (bytes memory) {
```

Comment:
+1!

There may be ways we want to expand this in the future, but I think this is the perfect functionality that we need today!